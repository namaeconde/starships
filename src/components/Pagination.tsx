import Button from "@mui/material/Button";
import { color } from "../theme";
import { makeStyles } from "@mui/styles";

export function paginate(array: Array<any>, pageSize: number, pageNumber: number): Array<any> {
  // human-readable page numbers usually start with 1, so we reduce 1 in the first argument
  return array.slice((pageNumber - 1) * pageSize, pageNumber * pageSize);
}

export default function Pagination({ onPreviousClick, onNextClick }: any): JSX.Element {
  const { paginationButton } = makeStyles(() => ({
    paginationButton: {
      fontSize: "24px",
      backgroundColor:`${color.racing_green}`,
      borderRadius:"20px",
      textAlign: "center",
    }
  }))();

  return (
    <>
      <Button className={paginationButton} color="secondary" sx={{ mx: 1 }} 
        onClick={onPreviousClick}
        disabled={onPreviousClick ? false : true }
      >
        Previous page
      </Button>
      <Button className={paginationButton} color="secondary" sx={{ mx: 1 }} 
        onClick={onNextClick}
        disabled={onNextClick ? false : true }
      >
        Next page
      </Button>
    </>
  )
}