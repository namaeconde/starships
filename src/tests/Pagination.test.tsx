import React from 'react';
import { render, cleanup } from '@testing-library/react'
import Pagination from '../components/Pagination';

afterEach(cleanup);

test('Pagination renders', () => {
  const { asFragment } = render(<Pagination />,);
  expect(asFragment()).toMatchSnapshot();
});
