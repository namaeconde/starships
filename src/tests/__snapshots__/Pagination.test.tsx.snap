// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders 1`] = `
<DocumentFragment>
  <button
    class="MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root Mui-disabled makeStyles-paginationButton-1 css-1oaij1c-MuiButtonBase-root-MuiButton-root"
    disabled=""
    tabindex="-1"
    type="button"
  >
    Previous page
  </button>
  <button
    class="MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root Mui-disabled makeStyles-paginationButton-1 css-1oaij1c-MuiButtonBase-root-MuiButton-root"
    disabled=""
    tabindex="-1"
    type="button"
  >
    Next page
  </button>
</DocumentFragment>
`;
